plugin/src/main/groovy/com/novoda/staticanalysis/internal/detekt/DetektConfigurator.groovy

```diff
@@ -0,0 +1,80 @@
+package com.novoda.staticanalysis.internal.detekt
+
+import com.novoda.staticanalysis.StaticAnalysisExtension
+import com.novoda.staticanalysis.internal.Configurator
+import com.novoda.staticanalysis.internal.Violations
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+class DetektConfigurator implements Configurator {
+
+    private final Project project
+    private final Violations violations
+    private final Task evaluateViolations
+
+    static DetektConfigurator create(Project project,
+                                     NamedDomainObjectContainer<Violations> violationsContainer,
+                                     Task evaluateViolations) {
+        Violations violations = violationsContainer.maybeCreate('Detekt')
+        return new DetektConfigurator(project, violations, evaluateViolations)
+    }
+
+    private DetektConfigurator(Project project, Violations violations, Task evaluateViolations) {
+        this.project = project
+        this.violations = violations
+        this.evaluateViolations = evaluateViolations
+    }
+
+    @Override
+    void execute() {
+        project.extensions.findByType(StaticAnalysisExtension).ext."detekt" = { Closure config ->
+
+            if (!isKotlinProject(project)) {
+                return
+            }
+
+            project.apply plugin: 'io.gitlab.arturbosch.detekt'
+
+            project.extensions.findByName('detekt').with {
+                // apply configuration closure to detekt extension
+                config.delegate = it
+                config()
+            }
+
+            if (project.tasks.findByName('detektCheck')) {
+                configureToolTask()
+            }
+        }
+    }
+
+    private void configureToolTask() {
+        def detektTask = project.tasks.findByName('detektCheck')
+        // run detekt as part of check
+        project.tasks.findByName('check').dependsOn(detektTask)
+
+        // evaluate violations after detekt
+        detektTask.group = 'verification'
+        def collectViolations = createCollectViolationsTask(violations)
+        evaluateViolations.dependsOn collectViolations
+        collectViolations.dependsOn detektTask
+    }
+
+    private CollectDetektViolationsTask createCollectViolationsTask(Violations violations) {
+        def reportFilePath = "${project.extensions.findByName('detekt').systemOrDefaultProfile().output}/detekt-checkstyle.xml"
+
+        project.tasks.create("collectDetektViolations", CollectDetektViolationsTask) { collectViolations ->
+            collectViolations.xmlReportFile = new File(reportFilePath)
```

Comment:
In case we have multiple profiles, can we have multiple `xmlReportFile`? 
I think `detekt` supports multiple profiles.

Reply:
Good point. Will add a JIRA task to follow-up on that 👍

Reply:
Since `profiles` is private, we cannot have multiple reports, right?

Reply:
Imho it's possible. The extension has a method to add profiles `fun profile(name: String, configuration: Action<in ProfileExtension>)`

```code
detekt { 
        profile('foo') { 
            output = "foo/reports"
        }

       profile('bar') { 
            output = "bar/reports"
        }
    }
```

Reply:
it is possible but how do you access them since `profiles` is private. I'm not sure if there is another way.

Reply:
Well, that's the challenge ^^